great_expectations/profile/user_configurable_profiler.py

```diff
@@ -0,0 +1,1122 @@
+import datetime
+import decimal
+from typing import Iterable
+
+import numpy as np
+from dateutil.parser import parse
+
+from great_expectations.core import ExpectationSuite
+from great_expectations.core.expectation_configuration import ExpectationConfiguration
+from great_expectations.dataset.util import build_categorical_partition_object
+from great_expectations.exceptions import ProfilerError
+from great_expectations.profile.base import (
+    OrderedProfilerCardinality,
+    ProfilerCardinality,
+    ProfilerDataType,
+    ProfilerDataTypesWithMapping,
+    ProfilerSemanticTypes,
+    ProfilerTypeMapping,
+)
+from great_expectations.profile.basic_dataset_profiler import (
+    BasicDatasetProfilerBase,
+    logger,
+)
+
+
+class UserConfigurableProfiler:
+    # TODO: Confirm tolerance for every expectation
+
+    """
+    The UserConfigurableProfiler is used to build an expectation suite from a dataset. The expectations built are
+    strict - they can be used to determine whether two tables are the same.
+
+    The profiler may be instantiated with or without a config. The config may contain a semantic_types dict or not. Once
```

Comment:
See note below, just need to update this to capture the init signature with parameters instead of config dict.

Reply:
Great catch! I updated this EOD yesterday from a config dictionary to top-level arguments after some internal conversation, but hadn't yet updated the documentation.